### ISSUE TITLE
Sample driver code re-declares all implemented functions as `virtual`, omits `override`

### ISSUE THREAD

**Issue by jherico** [ValveSoftware/openvr#161]
In the driver sample you have implementations of the various interfaces, like so

```
class CServerDriver_Sample: public IServerTrackedDeviceProvider
{
public:
	virtual EVRInitError Init( IDriverLog *pDriverLog, 	vr::IServerDriverHost *pDriverHost, const char *pchUserDriverConfigDir, const char *pchDriverInstallDir ) ;
	virtual void Cleanup() ;
        ...
```

However declaring the implemented methods like this is undesirable.  The methods are already virtual by virtue of inheriting `IServerTrackedDeviceProvider`, so the `virtual` keyword serves no purpose.  However, they should all be marked as `override` to indicate to the compiler that they are implementations of virtual functions.  

This has the advantage of generating a compiler error if the function signature changes in the base class but is not updated in a derived class.

**Comment by JoeLudwig** [comment]
You're right that the virtual keyword is unnecessary, it's just a useful hint to old guys like me that these are functions that come from a base class.

We use a #define (OVERRIDE) in the internal drivers that's defined as __override on Windows and nothing everywhere else. Some of this code (and that define) pre-date C++11 being widely available, so we couldn't just use "override". I actually stripped out the OVERRIDE references when I put together the sample so I could avoid including the header it's defined in.

I'll look into whether that's changed on the other platforms we care about and see if we can switch over. It would certainly make the sample better. (I've spent too many hours tracking down bugs caused by an override not getting called on a child class to want to ever do that again.)
